app/views/unattended/provisioning_templates/registration/linux_registration_default.erb

```diff
@@ -20,6 +20,19 @@ set -e
 <%= snippet 'puppet_setup' %>
 <% end -%>
 
+<% if host_param_true?('host_registration_insights') -%>
+echo '#'
+echo '# Installing Insights client'
+echo '#'
+
+yum install insights-client
+insights-client --register
```

Comment:
Be aware that if this fails for whatever reason, the script execution halts here (due to `set -e` earlier). That means that REX is not set up either so it's not easy to recover. That's why I would set up REX before insights.